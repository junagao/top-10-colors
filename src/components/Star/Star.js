import React from 'react'
import { bool, string, number, func } from 'prop-types'
import StarIcon from './Star.styles'

const Star = ({ id, index, full, onRate }) => {
  const ratingValue = index + 1

  return (
    <StarIcon
      size={20}
      color={full ? `#ffc107` : `#e4e5e9`}
      onClick={() => onRate(id, ratingValue)}
    />
  )
}

Star.propTypes = {
  id: string.isRequired,
  index: number.isRequired,
  full: bool.isRequired,
  onRate: func.isRequired,
}

export default Star
