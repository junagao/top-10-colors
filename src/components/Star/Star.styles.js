import styled from '@emotion/styled'
import { AiFillStar } from 'react-icons/ai'

const StarIcon = styled(AiFillStar)`
  cursor: pointer;
  transition: color 0.2s;
`

export default StarIcon
