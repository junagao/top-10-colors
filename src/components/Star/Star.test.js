import React from 'react'
import { shallow } from 'enzyme'
import Star from './Star'

describe('Star', () => {
  let wrapper

  const idMock = 'fc352593-c91d-4cf6-b689-7478ee2b61c8'
  const onRateMock = jest.fn()
  const indexMock = 4

  beforeEach(() => {
    wrapper = shallow(
      <Star id={idMock} index={indexMock} full onRate={onRateMock} />,
    )
  })

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render one StarIcon element', () => {
    expect(wrapper.find('StarIcon').length).toEqual(1)
  })

  it('should test click event on star and call onRate one time and pass `id` and `rating`', () => {
    wrapper.find('StarIcon').simulate('click')
    expect(onRateMock.mock.calls.length).toEqual(1)
    expect(onRateMock.mock.calls[0][0]).toEqual(
      'fc352593-c91d-4cf6-b689-7478ee2b61c8',
    )
    expect(onRateMock.mock.calls[0][1]).toEqual(5)
  })
})
