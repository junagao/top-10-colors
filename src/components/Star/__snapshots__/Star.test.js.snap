// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star should render correctly given the required props 1`] = `
<StarIcon
  color="#ffc107"
  onClick={[Function]}
  size={20}
/>
`;
